src/Controller/QuizInstanceController.php

```diff
@@ -56,16 +66,22 @@ public function showCandidateQuizzes(Request $request, array $requestAttributes)
         if ($redirectToLogin) {
             return $redirectToLogin;
         }
+        $orderBy = $this->quizInstanceService->getFromParameter('orderBy', $request, "");
+        $sortType = $this->quizInstanceService->getFromParameter('sort', $request, "");
         //TODO remove cast and modify methods to return the expected type
         $currentPage = (int)$this->quizInstanceService->getFromParameter('page', $request, 1);
         $totalResults = (int)$this->quizInstanceService->getEntityNumberOfPagesByField(QuizTemplate::class, []);
-        $quizzes = $this->quizInstanceService->getEntitiesByField(QuizTemplate::class, [], $currentPage, self::RESULTS_PER_PAGE);
+        $quizzes = $this->repoManager->getRepository(QuizTemplate::class)->getFilteredEntities(
+            new Filter([], self::RESULTS_PER_PAGE, ($currentPage - 1) * self::RESULTS_PER_PAGE, $orderBy, $sortType)
+        );
         $paginator = new Paginator($totalResults, $currentPage, self::RESULTS_PER_PAGE);
 
```

Comment:
why not pass the filter to the template? It already has the orderBy and sortType grouped together.